wgpu-core/src/device/resource.rs

```diff
@@ -2316,7 +2341,12 @@ impl<A: HalApi> Device<A> {
         let bgl_vec = desc
             .bind_group_layouts
             .iter()
-            .map(|&id| &try_get_bind_group_layout(bgl_guard, id).unwrap().raw)
+            .map(|&id| {
+                &try_get_bind_group_layout(bgl_guard, id)
+                    .unwrap()
+                    .assume_deduplicated()
+                    .raw
+            })
```

Comment:
This refactoring makes it hard to do the wrong thing because in most places where we interact with the BGL, we want to reach into some data that is now stored in the inner struct so you pretty much have to check whether it's a duplicate and follow the indirection to achieve that.